### ISSUE TITLE
Prevent layout from getting re-mounted on page change

### ISSUE THREAD

**Issue by futurfutur** [zeit/next.js#9881]
## Describe the bug
Trying Next with layout pattern:

https://github.com/zeit/next.js/tree/canary/examples/layout-component

And the problem is that Layout component get remounted on every page change. I need to use layout component as a Container so it'll fetch data from server on every mount. How can I prevent layout to get re-mounted? Or am I missing something there?

## To Reproduce
1. Download https://github.com/zeit/next.js/tree/canary/examples/layout-component
2. Add following code to Layout component:
```
React.useEffect(() => {
    console.log("layount mounted " + new Date());
  }, []);

```
3. See how Layout get remounted on every page, even though it's not supposed to.

## System information
MacOS
Chrome
Next.js latest

## Additional context
I've found a workaround in this article:
https://adamwathan.me/2019/10/17/persistent-layout-patterns-in-nextjs/

So maybe it's a feature request partly.

**Comment by JerryGreen** [comment]
The behaviour is indeed unpredictable. Though, there is that workaround (mentioned in the article), - and it's good enough. Still, the behaviour is different to an SPA, and it makes a lot of confusion since there are no foreseeable reasons for that. +1

**Comment by alabobriggs** [comment]
Consider adding the fetch call in `_app.js` and use a context to pass the data.

**Comment by futurfutur** [comment]
> Consider adding the fetch call in `_app.js` and use a context to pass the data.

What if I need to implement multiple Containers? I can put all fetch requests to _app.js and pass them as an api via context to child components. But still how could I call an api for each layout-container only once  (on the first mount?)

**Comment by timneutkens** [comment]
Duplicate of #8193

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
